### PR TITLE
MGMT-11362: Set api vip and not dnsname for day2 cluster (transform) for bm platform

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-11362](https://issues.redhat.com//browse/MGMT-11362): Set api vip and not dnsname for day2 cluster (transform) for bm platform
We see cases where user are not able to manage dns inside their org and
they are stuck with trying to add hosts. We know that in BM case there
is no need to set dns name if we can set api vip.
Changing tranform to day2 logic to set vip in case it can

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/common/common.go [hunk 1]
```diff
@@ -393,6 +393,16 @@ func GetTagFromImageRef(ref string) string {
 	}
 }
 
+func GetAPIVipDNSName(c *Cluster) string {
```
Comment (eranco74) [inline internal/common/common.go:396]: Consider renaming to GetIgnitionEndpoint?

File: internal/common/common.go [hunk 2]
```diff
@@ -393,6 +393,16 @@ func GetTagFromImageRef(ref string) string {
...
+	//In case cluster that is not None platform and api vip is set we should set api vip as APIVipDNSName
+	// I know that APIVipDNSName name is confusing but in order not to add new variable we will use it
+	// as we allow already setting ip to it
+	if !swag.BoolValue(c.Cluster.UserManagedNetworking) && c.Cluster.APIVip != "" {
```
Comment (eranco74) [inline internal/common/common.go:400]: Unsure we can trust UMN, but I guess if we have the API VIP we can just use it

File: internal/cluster/cluster_test.go [hunk 3]
```diff
@@ -2905,82 +2905,124 @@ var _ = Describe("Transform day1 cluster to a day2 cluster", func() {
...
-		name          string
-		clusterStatus string
-		clusterKind   string
-		errorExpected bool
+		name               string
+		clusterStatus      string
+		clusterKind        string
+		errorExpected      bool
+		userManagedNetwork bool
+		apiVip             string
```
Comment (omertuc) [inline internal/cluster/cluster_test.go:2913]:
```suggestion
		expectedAPIVipDNSName             string
```

File: internal/common/common.go (hunk 1 above)
Comment (omertuc) [reply internal/common/common.go:396]: I disagree, the literal APIVipDNSName cluster field gets set with the value being the string returned by this function

Comment (omertuc) [inline internal/common/common.go:396]: Perhaps this function should be called "GetConvertedClusterAPIVipDNSName", because it assumes UMN is correctly set, which is only true during conversion

File: internal/common/common.go (hunk 2 above)
Comment (tsorya) [reply internal/common/common.go:400]: i prefer to verify both, in SNO case we will set api_vip but we still need to use dns name

File: internal/cluster/cluster_test.go (hunk 3 above)
Comment (tsorya) [reply internal/cluster/cluster_test.go:2913]: this should be ip as i need  away to verify case where apiVip is not set

Comment (omertuc) [reply internal/cluster/cluster_test.go:2913]: The test is literally just checking whether the APIVipDNSName field on the cluster is equal to this parameter, so the name of this parameter should be modified to emphasize that it's an expectation (i.e. "test output") and not a test "input" like the rest of the parameters in this struct

Comment (omertuc) [reply internal/cluster/cluster_test.go:2913]: Ah sorry I misread the test

File: internal/cluster/cluster_test.go [hunk 4]
```diff
@@ -3007,8 +3050,13 @@ var _ = Describe("Transform day1 cluster to a day2 cluster", func() {
 				Expect(db.Take(&c, "id = ?", cluster.ID.String()).Error).ToNot(HaveOccurred())
 				Expect(c.Kind).To(Equal(swag.String(models.ClusterKindAddHostsCluster)))
 				Expect(c.Status).To(Equal(swag.String(models.ClusterStatusAddingHosts)))
-				apiVipDnsname := fmt.Sprintf("api.%s.%s", c.Name, c.BaseDNSDomain)
-				Expect(c.APIVipDNSName).To(Equal(swag.String(apiVipDnsname)))
+				if t.apiVip == "" || t.userManagedNetwork {
+					apiVipDnsname := fmt.Sprintf("api.%s.%s", c.Name, c.BaseDNSDomain)
+					Expect(c.APIVipDNSName).To(Equal(swag.String(apiVipDnsname)))
+				} else {
+					Expect(c.APIVipDNSName).To(Equal(swag.String(t.apiVip)))
```
Comment (omertuc) [inline internal/cluster/cluster_test.go:3057]: I think this demonstrates the intention better:

```suggestion
					Expect(c.APIVipDNSName).To(Equal(swag.String(cluster.APIVip)))
```

File: internal/cluster/cluster_test.go (hunk 3 above)
Comment (omertuc) [reply internal/cluster/cluster_test.go:2913]: See https://github.com/openshift/assisted-service/pull/4191/files#r931196094 instead, it's the reason I got confused

File: internal/cluster/cluster_test.go (hunk 4 above)
Comment (tsorya) [reply internal/cluster/cluster_test.go:3057]: sounds good

File: internal/common/common.go [hunk 5]
```diff
@@ -393,6 +393,16 @@ func GetTagFromImageRef(ref string) string {
 	}
 }
 
+func GetConvertedClusterAPIVipDNSName(c *Cluster) string {
+	//In case cluster that is not None platform and api vip is set we should set api vip as APIVipDNSName
+	// I know that APIVipDNSName name is confusing but in order not to add new variable we will use it
```
Comment (eranco74) [inline internal/common/common.go:398]: I don't see `APIVipDNSName` here, this comment seems unrelated

File: internal/common/common.go [hunk 6]
```diff
@@ -393,6 +393,16 @@ func GetTagFromImageRef(ref string) string {
 	}
 }
 
+func GetConvertedClusterAPIVipDNSName(c *Cluster) string {
+	//In case cluster that is not None platform and api vip is set we should set api vip as APIVipDNSName
```
Comment (eranco74) [inline internal/common/common.go:397]:
```suggestion
	//In case cluster that isn't configred with user-managed-networking and api vip is set we should set api vip as APIVipDNSName
```

**Review by eranco74 — COMMENTED** [review]
/lgtm
